### PR TITLE
Remove empty entry for `OrphanPool`

### PR DESCRIPTION
**Pull request by eval-exec**
### What problem does this PR solve?
When an entry in `OrphanPool` is empty, it should be removed.

### What is changed and how it works?

What's Changed:

### Related changes

- Remove empty entry in `OrphanPool`

### Check List 

Tests 

- Unit test
- Integration test
- Manual test (add detailed scripts or steps below)
- No code ci-runs-only: [ quick_checks,linters ]

Side effects

- None

### Release note 

```release-note
Title Only: Include only the PR title in the release note.
```

### CODE REVIEW

File: tx-pool/src/process.rs [hunk 1]
```diff
@@ -751,6 +750,10 @@ impl TxPoolService {
 
         if verify_cache.is_none() {
             // update cache
+            let TransactionView {
```
Comment (doitian) [inline tx-pool/src/process.rs:753]: how about just `drop(tx)`;

Comment (eval-exec) [reply tx-pool/src/process.rs:753]: OK, I will try `drop(tx)`.

File: tx-pool/src/component/orphan.rs [hunk 2]
```diff
@@ -78,6 +78,11 @@ impl OrphanPool {
                 self.by_out_point
                     .get_mut(&out_point)
                     .map(|set| set.remove(id));
+
+                // remove empty by_out_point entry
```
Comment (doitian) [inline tx-pool/src/component/orphan.rs:82]: I think you should test `self.by_out_point[out_point]` is empty.

Comment (eval-exec) [reply tx-pool/src/component/orphan.rs:82]: Yes, I made a mistake.

**Review by chenyukang — DISMISSED** [review]

File: util/types/src/core/views.rs [hunk 3]
```diff
@@ -298,6 +298,15 @@ macro_rules! define_inner_getter {
 }
 
 impl TransactionView {
+    /// Destruct TransactionView to get wtx_hash
+    pub fn destruct_wtx_hash(self) -> packed::Byte32 {
+        let TransactionView {
+            witness_hash: wtx_hash,
+            ..
+        } = self;
+        wtx_hash
```
Comment (doitian) [inline util/types/src/core/views.rs:307]: Not sure but `self.wtx_hash` should just work

Comment (doitian) [inline util/types/src/core/views.rs:307]: Not sure but `self.witness_hash` should just work

```suggestion
        self.witness_hash
```

Comment (eval-exec) [reply util/types/src/core/views.rs:307]: Not sure too. @quake

Comment (eval-exec) [reply util/types/src/core/views.rs]: We have confirmed that holding a witness_hash will not prevent the release of TransactionView.

**Review by zhangsoledad — APPROVED** [review]

**Pull request merged**